mimic/canned_responses/auth.py

```diff
@@ -81,6 +113,10 @@ def endpoint_json():
         }
     }
 
+    if (entry_generator and prefix_for_endpoint):
```

Comment:
Anywhere that you use a raw object in an `if` statement, you're calling its `__nonzero__` method, which can have side-effects.  If the only thing you want to test about an object is identity with `None`, it should just be `is None`, since that's a tighter contract; it's more specific.  So I think this should be `if entry_generator is not None and prefix_for_endpoint is not None`.